internal/clusterfeature/clusterfeatureadapter/cadence_feature_operation_dispatcher.go

```diff
@@ -27,50 +27,39 @@ import (
 	"github.com/banzaicloud/pipeline/internal/common"
 )
 
-type asyncFeatureManagerStub struct {
-	clusterfeature.FeatureManager
-	cadenceClient     client.Client
-	featureRepository clusterfeature.FeatureRepository
-	logger            common.Logger
-}
-
-// NewAsyncFeatureManagerStub returns a new, asynchronous feature manager stub
-func NewAsyncFeatureManagerStub(
-	featureManager clusterfeature.FeatureManager,
-	featureRepository clusterfeature.FeatureRepository,
+// MakeCadenceFeatureOperationDispatcher returns an Uber Cadence based implementation of FeatureOperationDispatcher
+func MakeCadenceFeatureOperationDispatcher(
 	cadenceClient client.Client,
 	logger common.Logger,
-) clusterfeature.FeatureManager {
-	return asyncFeatureManagerStub{
-		FeatureManager:    featureManager,
-		cadenceClient:     cadenceClient,
-		featureRepository: featureRepository,
-		logger:            logger,
+) CadenceFeatureOperationDispatcher {
```

Comment:
```suggestion
) *CadenceFeatureOperationDispatcher {
```

Reply:
I think that this is unnecessary. As a performance optimization it would be premature. As for instance sharing, you could make that decision on the caller's side by taking the address of the returned instance and passing that around. Also, instance sharing is only relevant if you mutate the instance, which this implementation does not do. See also: https://stackoverflow.com/a/23551970/578530